inc/class-wpseo-content-images.php

```diff
@@ -70,10 +70,9 @@ private function get_images_from_content( $content ) {
 		foreach ( $this->get_img_tags_from_content( $content ) as $img ) {
 			$url = $this->get_img_tag_source( $img );
 			if ( $url ) {
-				$images[ $url ] = WPSEO_Image_Utils::get_attachment_by_url( $url );
+				$images[ $url ] = $url;
```

Comment:
If we just change this from an Attachment ID to an URL, then the cache that exists on installations will not have the same structure and will produce unwanted behaviour.

This is fine, if we also introduce an upgrade routine that clears out all the currently stored caches from the post_metas.